AI #2334: Align remaining normative requirement inconsistencies with updated guidelines (#2344)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/cost_and_usage/columns/tags.md

  * Tags SHOULD include tag keys with corresponding non-null values for a given [*resource*](#glossary:resource).
  * Tags MAY include tag keys with a null value for a given *resource* depending on the data generator's tag finalization process.
  * Tag keys that do not support corresponding values, MUST have a corresponding true (boolean) value set.

>   * Tag values MUST match the original values unless true (boolean) is applied to valueless tags.

Author: @kk09v, following [this ](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/2344/files#r3203155401)suggestion, you'd prefer this, right?

```suggestion
  * Tag values MUST match the provided values unless true (boolean) is applied to valueless tags.
```

* Provider-defined tags MUST adhere to the following requirements:
  * Provider-defined tag keys MUST be prefixed with a predetermined, provider-specified tag key prefix that is unique to each corresponding provider-specified [*tag scheme*](#glossary:tag-scheme).
  * Provider-specified tag key prefixes SHOULD be publicly documented.
* User-defined tags MUST adhere to the following requirements:
  * User-defined tag keys in all but one user-defined *tag scheme* MUST include a predetermined, provider-specified tag key prefix that is unique to each corresponding user-defined *tag scheme* when the data generator has more than one user-defined *tag scheme*.
  * User-defined tag keys MUST NOT include a *tag scheme*-specific prefix when the data generator has only one user-defined *tag scheme*.

>   * Reserved tag key prefixes MUST NOT be used as prefixes for any user-defined tag keys within a prefixless user-defined *tag scheme*.

Reviewer: This line I believe changes the meaning. By removing the data generator as the party responsible for preventing this, there is ambiguity. This line could now be interpreted as telling practitioners that they MUST NOT use a reserved tag key prefix when they are applying tag keys.

Author: option A) revert only this requirement

```suggestion
  * Data generator MUST NOT allow reserved tag key prefixes to be used as prefixes for any user-defined tag keys within a prefixless user-defined *tag scheme*.
```

option B) revert all updates to this column

Reviewer: ```suggestion
  * Reserved tag key prefixes MUST be prevented from being used as prefixes for any user-defined tag keys within a prefixless user-defined *tag scheme*.
```
Here's an attempt.

Reviewer: Alternatively, we could change the subject
```suggestion
  * Tagging capabilities MUST prevent reserved tag prefixes from being used for any user-defined tag keys within a prefixless user-defined *tag scheme*.
```

Reviewer 2: Discussed with @ijurica.  Will accept @kk09v suggestion.
